File: back-end/src/main/java/javaweb/my_project/entities/MarketPrice.java
package javaweb.my_project.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class MarketPrice {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    Integer price;

    String region;

    LocalDateTime dateRecorded;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    Product product;
}

File: front-end/src/types/market-price.ts
import { IProductMarketPriceResponse } from "@/types/product";

export interface IMarkerPriceResponse {
  id: string;
  price: number;
  dateRecorded: string;
  region: string;
  product: IProductMarketPriceResponse;
}

export interface IMarketPriceCreationRequest {
  price: number;
  dateRecorded: string;
  region: string;
  productId: string;
}

export interface IMarketPricPatchUpdateRequest {
  price?: number;
  dateRecorded?: string;
  region?: string;
}
